### ISSUE TITLE
macOS build error on 4.0.0 : undeclared identifier 'CLOCK_MONOTONIC'

### ISSUE THREAD

**Issue by jakirkham** [pyutils/line_profiler#185]
On macOS builds am seeing the following error:

```C
line_profiler/timers.c:64:23: error: use of undeclared identifier 'CLOCK_MONOTONIC'
```

This traces back to this line:

https://github.com/pyutils/line_profiler/blob/623078eab1b8733efb7d328a31fc89172feeb5eb/line_profiler/timers.c#L64

More details in [this CI log]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=611334&view=logs&jobId=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=20c71c51-4b27-578b-485d-06ade2de1d00 ). Have also [attached the log for posterity]( https://github.com/pyutils/line_profiler/files/10020778/line_profiler-feedstock_20221112.1_osx_64_python3.10.____cpython.txt )

**Comment by jakirkham** [comment]
Think we may need something like [this]( https://github.com/conda-forge/compiler-rt-feedstock/blob/b0c99fa0dec7f3b0d96bf8e99ff6b069d28abe0c/recipe/macosx_10_9.patch#L47 )

**Comment by Theelx** [comment]
Ah you're right, we forgot to test Mac when adding the nanosecond resolution timers. I appreciate the link to the conda-forge fix, I'll try to implement that in a PR today.

**Comment by Erotemic** [comment]
@Theelx the CI does test on OSX. Not sure why ours work, but the conda ones fail.

**Comment by Theelx** [comment]
Do we know what versions of OSX are being used for our pipeline vs conda's?

**Comment by jakirkham** [comment]
Conda-forge is building with an older macOS SDK (10.9) to support a wider range of users. Sounds like [wheels are doing the same thing]( https://github.com/pypa/cibuildwheel#what-does-it-do ). Admittedly that is getting to be a pretty old macOS version.

Looking at the patch linked above, it appears this feature isn't included until 10.12, which is more recent.

**Comment by henryiii** [comment]
You can request a newer macOS for a conda-recipe, we have to do that for root-feedstock and I think some others. 10.9 is _really_ old at this point, is conda-forge ever going to update that?

**Comment by jakirkham** [comment]
Feel free to share feedback in issue ( https://github.com/conda-forge/conda-forge.github.io/issues/1844 )

In any event for the scope of this issue, agree we can bump the minimum OS requirement in conda-forge, which I'm now doing in the feedstock

Not sure how that is handled for wheels given they would have the same issue. Perhaps folks here have thoughts on that 🙂

**Comment by henryiii** [comment]
I'd personally recommend just bumping to 10.12 for wheels too. 10.12 is plenty old at this point. That's the first version that starts having some C++17 support (10.14 for full C++17 support). CMake's wheels bumped from 10.9 to to 10.10 and no one has complained yet. :)

**Comment by henryiii** [comment]
Yes, you set `MACOSX_DEPLOYMENT_TARGET` to 10.12. We will support 10.9+ in cibuildwheel until CPython raises the minimum they compile with. But it's fine to increase that for a package.
